src/lib/RoutingManager.ts

```diff
@@ -36,7 +36,7 @@ class RoutingManager implements Widget {
     this.currentUiState = this.stateMapping.routeToState(this.router.read());
   }
 
-  private getAllSearchParameters({
+  public getAllSearchParameters({
```

Comment:
In the context of this PR, are you changing the accessibility of the methods just for the tests? If yes, I do not think this is a good idea :neutral_face:  . We should limit our tests to (real) public methods and potential side effects and not trying to test internal state.

WDYT?

Reply:
Yes, that's why I changed the visibility. I don't think that's good either but I didn't want to dramatically update the tests/implementation because of the migration.

Do you have an idea in mind how to rewrite the test while keeping the private visibility? Maybe this doesn't need to be tested at this level.

Reply:
Since these tests are testing specifically the `getAllSearchParameters` method I would have removed them. The conditions that are tested here should ideally be covered by the other tests, but I don't know if it is the case :thinking: 

In doubt, for now I would have added a `// @ts-ignore` comment in the tests, not perfect but I find this better than publicly exposing something that should not be used.

WDYT?

Reply:
That's indeed an acceptable solution for now!